lib/open_mic.rb

```diff
@@ -0,0 +1,18 @@
+require './lib/joke'
+require './lib/user'
+
+class OpenMic
+attr_reader :location, :date, :performers
+
+  def initialize(attributes = {})
+    attributes.each do |attr, value|
+    define_singleton_method("#{attr}=") { |val| location[attr] = val }
+    define_singleton_method(attr) { attributes[attr] }
+    end
```

Comment:
This works okay, but (a) indentation is harder to read, and (b) this is over-complicated when there are only two attributes we wanted you to track. If we were tracking 100 things in a hash, then this would be a good choice.